cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -422,8 +422,15 @@ func (s *Server) GetInstalledPackageSummaries(ctx context.Context, request *core
 			pkgDataForNamespaces := pkgDatas[pkgName]
 			// Check if there was package metadata for the specific namespace.
 			pkgData := pkgDataForNamespaces[pkgi.Namespace]
+			var ok bool
 			if pkgData.meta == nil {
-				pkgData = pkgDataForNamespaces[s.globalPackagingNamespace]
+				pkgData, ok = pkgDataForNamespaces[s.globalPackagingNamespace]
+				// Ignore packages which do not have associated metadata
+				// available. See https://github.com/vmware-tanzu/kubeapps/issues/4901
+				if !ok || pkgData.meta == nil {
```

Comment:
> So here we're ignoring the whole package in the GetPkgSummaries response, which seems fair enough (1). However, correct me if I'm wrong here, the package will still be retrievable using the `GetPkgDetail` endpoint (even if manually accessing via API or using the UI) hardcoding the URL)

Would it? I've not tested it, but *thought* it would 404 in this case due to:

https://github.com/vmware-tanzu/kubeapps/blob/28f0e1512604800a9b3bfc63c01a1d176137fb94/cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go#L484-L487


> 
> Is it the desired behavior? Do we want to display details even if they are not presented in the summaries response ?

I don't see an issue if our API does, when this is a work-around to an error. I'm more than happy for a better long-term solution for invalid packages (such as returning them in the summaries but as invalid or with some indication that info is missing etc.).

> 
> (1) Even if hiding the invalid packages is a legit action... we would be differing from a plain `kubectl get pkgi` , which may seem confusing for some users. What about adding a config flag to defer this decision to the kubeapps operator? I mean, populating those fields coming from a pkgmetadata with a `unknown` string would be enough, maybe.

I don't think differing with kubectl here is an issue, but we would want to check if we differ from what the tanzu cli does here for getting installed packages.

Reply:
> Is it the desired behavior? Do we want to display details even if they are not presented in the summaries response ?

If that is the case, it might be confusing for users. I'm more inclined to a more coherent solution like Michael's mention of `such as returning them in the summaries but as invalid or with some indication that info is missing etc.`. So that all packages are returned by APIs (summary and detail) and Kubeapps is not hiding any. Those invalid can be marked with some new flag if they can't be installed due to missing data/metadata perhaps?